### PR TITLE
[chore] Update GitHub Actions to use Node 20

### PR DESCRIPTION
**Pull request by heatlikeheatwave**
## 🔗 Relevant links

- [Asana task](https://app.asana.com/0/0/1206490176997278/f) 🎟️

## 🗒️ What

- Updated the GHA workflows to use Node version 20 and NPM v10 instead of nvmrc
- Updated Node to v9.6.7 to match nvmrc for in the repo's `package.json`
- Updated the following actions to use Node.js 20:
  - [x] actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9,
  - [x] actions/setup-node@e33196f7422957bea03ed53f6fbb155025ffc7b8,
  - [x] bahmutov/npm-install@c5482d70ec8706408996e31ac94075030694993f,
  - [x] actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8,
  - [x] actions/stale@1160a2240286f5da8ec72b1c0816ce2481aabf84,
  - [x] actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce,
  - [x] docker/setup-qemu-action@2b82ce82d56a2a04d2637cd93a637ae1b359c0a7,
  - [x] docker/setup-buildx-action@2a1a44ac4aa01993040736bd95bb470da1a38365,
  - [x] docker/login-action@465a07811f14bebb1938fbed4728c6a1ff8901fc,
  - [x] docker/metadata-action@818d4b7b91585d195f67373fd9cb0332e31a7175,
  - [x] docker/build-push-action@2eb1c1961a95fc15694676618e422e8ba1d63825,

## 🤷 Why

Node 16 has reached its [end of life](https://github.com/nodejs/Release/#end-of-life-releases), prompting GitHub to initiate its deprecation process for GitHub Actions. Their plan is to transition all actions to run on Node 20 by Spring 2024 ([source](https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/))

## 🛠️ How

- Used specific commits SHAs from [HashiCorp's Trusted Supply Chain Component Registry (TSCCR) ](https://github.com/hashicorp/security-tsccr/tree/main/components/github_actions)

## 🧪 Testing

- Check the GHA and see there are no failures or deprecation notices like the one below:
deprecation

## 💭 Anything else?

nay

### CODE REVIEW

File: .github/workflows/nextjs_bundle_analysis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
         uses: bahmutov/npm-install@c5482d70ec8706408996e31ac94075030694993f # v1.8.32
```
Comment (EnMod) [inline .github/workflows/nextjs_bundle_analysis.yml:32]: 🤔 It seems this dep might need an update as well, according to this check log:
https://github.com/hashicorp/dev-portal/actions/runs/7889328354

**Review by EnMod — COMMENTED** [review]
LGTM for the most part! Just had a warning and small edit to make.

File: .github/workflows/nextjs_bundle_analysis.yml (hunk 1 above)
Comment (heatlikeheatwave) [reply .github/workflows/nextjs_bundle_analysis.yml]: Good catch! Corrected here 533820d67116101334aa472908c92a1ff680531b

**Review by EnMod — APPROVED** [review]
All green! 🎉

**Pull request merged**